### PR TITLE
Add channel name to the results

### PR DESCRIPTION
**Pull request by spyrkob**
Part of wildfly-extras/prospero#411

Add the name of the channel the artifact is found in to the results returned by ChannelSession

### CODE REVIEW

File: core/src/main/java/org/wildfly/channel/VersionResult.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+/**
+ * The result of version search.
+ */
+public class VersionResult {
+    private String version;
+    private String channelName;
+
+    public VersionResult(String version, String channelName) {
+        Objects.requireNonNull(version);
+        Objects.requireNonNull(version);
```
Comment (jmesnil) [inline core/src/main/java/org/wildfly/channel/VersionResult.java:30]: duplicated line

File: core/src/main/java/org/wildfly/channel/MavenArtifact.java [hunk 2]
```diff
@@ -23,18 +23,33 @@
...
     public File getFile() {
         return file;
     }
 
+    /**
+     * The name of the channel the artifact was resolved from.
+     *
+     * @return name of the chanel or {@code null} if the artifact was resolved directly.
+     */
+    public String getChannelName() {
```
Comment (jmesnil) [inline core/src/main/java/org/wildfly/channel/MavenArtifact.java:49]: could you make it return `Optional<String>` instead?

File: core/src/main/java/org/wildfly/channel/VersionResult.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+     */
+    public String getVersion() {
+        return version;
+    }
+
+    /**
+     * The name of the channel the artifact was found in.
+     * @return
+     */
+    public String getChannelName() {
```
Comment (jmesnil) [inline core/src/main/java/org/wildfly/channel/VersionResult.java:49]: could you make it return `Optional<String>` instead?

**Review by jmesnil — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/wildfly/channel/VersionResult.java (hunk 3 above)
Comment (spyrkob) [reply core/src/main/java/org/wildfly/channel/VersionResult.java:49]: Does the Optional make sense for VersionResult? The `MavenArtifact` might be resolved using 'resolveDirectMavenArtifact' bypassing channels, but I think the version query would always have to be resolved against one of the channel

Comment (jmesnil) [reply core/src/main/java/org/wildfly/channel/VersionResult.java:49]: @spyrkob that makes sense. So I suppose that one of the `Objects.requireNonNull` above was for the `channelName`, right?

Comment (spyrkob) [reply core/src/main/java/org/wildfly/channel/VersionResult.java:49]: yes it was, I must have copy/pasted the line and forgot to change the parameter name, sorry

Comment (spyrkob) [reply core/src/main/java/org/wildfly/channel/VersionResult.java:49]: I realised that's not correct after all. The channel name is not mandatory in the specification, so in both `VersionResult` and `MavenArtifact` the channelName can be null even if the artifact/version was resolved from one.

I'll update the PR with the change

**Review by jmesnil — APPROVED** [review]

**Pull request merged**